server/service/src/invoice/outbound_return/delete/mod.rs

```diff
@@ -0,0 +1,211 @@
+use repository::{ActivityLogType, InvoiceRowRepository, RepositoryError, TransactionError};
+
+pub mod validate;
+
+use validate::validate;
+
+use crate::{
+    activity_log::activity_log_entry,
+    invoice::common::get_lines_for_invoice,
+    invoice_line::stock_out_line::{
+        delete::{delete_stock_out_line, DeleteStockOutLine, DeleteStockOutLineError},
+        StockOutType,
+    },
+    service_provider::ServiceContext,
+    WithDBError,
+};
+
+type OutError = DeleteOutboundReturnError;
+
+pub fn delete_outbound_returns(
+    ctx: &ServiceContext,
+    ids: Vec<String>,
+) -> Result<Vec<String>, OutError> {
+    ctx.connection
+        .transaction_sync(|connection| -> Result<(), OutError> {
+            for id in &ids {
+                validate(&id, &ctx.store_id, &connection)?;
+
+                let lines = get_lines_for_invoice(connection, &id)?;
+                for line in lines {
+                    delete_stock_out_line(
+                        ctx,
+                        DeleteStockOutLine {
+                            id: line.invoice_line_row.id.clone(),
+                            r#type: Some(StockOutType::OutboundReturn),
+                        },
+                    )
+                    .map_err(|error| OutError::LineDeleteError {
+                        line_id: line.invoice_line_row.id,
+                        error,
+                    })?;
+                }
+
+                activity_log_entry(
+                    &ctx,
+                    ActivityLogType::InvoiceDeleted,
+                    Some(id.to_owned()),
+                    None,
+                    None,
+                )?;
+
+                InvoiceRowRepository::new(&connection)
+                    .delete(&id)
+                    .map_err(|error| OutError::DatabaseError(error))?;
+            }
+            Ok(())
+        })
+        .map_err(|error| error.to_inner_error())?;
+
+    Ok(ids)
+}
+
+#[derive(Debug, PartialEq, Clone)]
+
+pub enum DeleteOutboundReturnError {
+    InvoiceDoesNotExist,
+    DatabaseError(RepositoryError),
+    NotThisStoreInvoice,
+    CannotEditFinalised,
+    LineDeleteError {
+        line_id: String,
+        error: DeleteStockOutLineError,
+    },
+    NotAnOutboundReturn,
+}
+
+impl From<RepositoryError> for DeleteOutboundReturnError {
+    fn from(error: RepositoryError) -> Self {
+        DeleteOutboundReturnError::DatabaseError(error)
+    }
+}
+
+impl From<TransactionError<DeleteOutboundReturnError>> for DeleteOutboundReturnError {
```

Comment:
Maybe implementing this would avoid the need to specify the return type?
```
impl From<TransactionError<DeleteOutboundReturnError>> for RepositoryError {
```
I don't think we need to do it though!